src/avif.c

```diff
@@ -1087,13 +1087,14 @@ avifCodecChoice avifCodecChoiceFromName(const char * name)
     return AVIF_CODEC_CHOICE_AUTO;
 }
 
-avifCodec * avifCodecCreate(avifCodecChoice choice, avifCodecFlags requiredFlags)
+avifResult avifCodecCreate(avifCodecChoice choice, avifCodecFlags requiredFlags, avifCodec ** codec)
```

Comment:
Is the idea to move APIs (like avifAlloc) to return avifResult? I could not agree more with the nodiscard I am adding.

Reply:
The reason here is that `avifCodecCreate()` can fail in multiple ways (no codec available or out of memory). `avifResult` is also convenient for the calling site.